File: Frontend/library-app/src/models/PaymentHistory.ts
class PaymentHistory {
  title: string;
  author: string;
  category: number;
  img: string;
  publisher: string;
  amount: number;
  price: number;
  paymentDate: string;
  orderId: string;
  constructor(
    title: string,
    author: string,
    category: number,
    img: string,
    publisher: string,
    amount: number,
    price: number,
    paymentDate: string,
    orderId: string
  ) {
    this.title = title;
    this.author = author;
    this.category = category;
    this.img = img;
    this.publisher = publisher;
    this.amount = amount;
    this.price = price;
    this.paymentDate = paymentDate;
    this.orderId = orderId;
  }
}
export default PaymentHistory;

File: src/main/java/com/reactlibraryproject/springbootlibrary/Entity/PaymentHistory.java
package com.reactlibraryproject.springbootlibrary.Entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.Email;

@Entity
@Table(name = "payment_history")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PaymentHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Email
    @Column(name = "user_email")
    private String userEmail;

    @Column(name = "title")
    private String title;

    @Column(name = "author")
    private String author;

    @Column(name = "category")
    private String category;

    @Column(name = "img")
    private String img;

    @Column(name = "publisher")
    private String publisher;

    @Column(name = "amount")
    private int amount;

    @Column(name = "price")
    private int price;

    @Column(name = "payment_key")
    private String paymentKey;

    @Column(name = "payment_date")
    private String paymentDate;

    @Column(name = "order_id")
    private String orderId;

    @Column(name = "status")
    private String status;

    @Column(name = "cart_item_id")
    private Long cartItemId;
}
